common/tea_cmd/git_test.go

```diff
@@ -0,0 +1,54 @@
+package tea_cmd
+
+import (
+	"fmt"
+	"os"
+	"testing"
+)
+
+const templateUrlTest = "https://github.com/Argus-Labs/starter-game-template.git"
+
+func TestGitCloneCmd(t *testing.T) {
+	type param struct {
+		url       string
+		targetDir string
+		initMsg   string
+	}
+
+	test := []struct {
+		name    string
+		wantErr bool
+		param   param
+	}{
+		{
+			name:    "error clone wrong address",
+			wantErr: true,
+			param: param{
+				url:       "wrong address",
+				targetDir: "targetDir",
+				initMsg:   "initMsg",
+			},
+		},
+		{
+			name:    "success",
+			wantErr: false,
+			param: param{
+				url:       templateUrlTest,
+				targetDir: os.TempDir() + "/worldclitest",
+				initMsg:   "initMsg",
+			},
+		},
+	}
+
+	for _, tt := range test {
+		err := GitCloneCmd(tt.param.url, tt.param.targetDir, tt.param.initMsg)
+		if (err != nil) != tt.wantErr {
+			t.Errorf("SaveCreationProgress() wantErr: %v but got: %v", tt.wantErr, err)
+		} else {
+			err = os.RemoveAll(tt.param.targetDir)
+			if err != nil {
+				fmt.Println(err)
+			}
+		}
+	}
+}
```

Comment:
i'd recommend using an assertion library https://pkg.go.dev/gotest.tools/assert

this way you can do 

```go
if tt.wantErr {
    assert.Error(t, err, "custom message if this assertion fails")
} else {
    assert.NilError(t, err)
}
```

Reply:
updated